### PR TITLE
Fix #75: Publish vsix files on release

### LINKED ISSUE #75: Artifacts not published to GitHub releases
The releases pages on GitHub, eg
https://github.com/scala/vscode-scala-syntax/releases/tag/0.3.9
https://github.com/scala/vscode-scala-syntax/releases/tag/0.1.0
don't contain the build artifact, like `scala-lang.scala-0.3.9.vsix` and `scala-lang.scala-snippets-0.1.0.vsix`, uploaded and viable there

the `vsix` files is only available via Microsoft's Marketplace
https://marketplace.visualstudio.com/_apis/public/gallery/publishers/scalameta/vsextensions/metals/1.7.2/vspackage

That is not a viable source for other IDEs.
The `scala-lang.scala` and `scala-lang.scala-snippets` `vsix` files have to be available somewhere else, GitHub releases is probably the best place, so that Metals is officially included in Eclipse Che web IDE.
https://github.com/eclipse/che-plugin-registry/pull/356

Related issue: https://github.com/scalameta/metals/issues/1365

### PR DESCRIPTION
**Pull request by anatoliykmetyuk**

### CODE REVIEW

**Comment by anatoliykmetyuk** [comment]
So far it seems that only `scala-lang.scala` visx file is generated. However, in the issue @sideeffffect mentioned also the `scala-lang.scala-snippets` vsix file. Does anyone know where exactly the latter comes from?

**Comment by sideeffffect** [comment]
Seems like `snippets` has been canceled: https://github.com/scala/vscode-scala-syntax/issues/18

btw, please make sure that the format of the name of the file is the same as on VS Marketplace, e.g. `scala-lang.scala-0.3.9.vsix` (and not something like `scala-0.3.9.vsix`)

:pray:

**Comment by anatoliykmetyuk** [comment]
Done. That deployment logic of Travis looks fishy though, I'm not sure if this will work. They required me to generate a GitHub API key which is generated on a per-profile basis. If that key is tied to my account, the publishing might fail because I don't have the write rights for this repo.

File: .travis.yml [hunk 1]
```diff
@@ -14,4 +14,14 @@ jobs:
     - stage: release
       script: bin/ci-release.sh
 
+deploy:
+  provider: releases
+  api_key:
+    secure: "Wt9gFsQqxwFOnLFuMgc6fy1d4Z4T2v3xfyhb4+bqip/MTUR3K9vwUAmvsxWe6/rPfFsdWb+Iou2VH5t3Bh5lck2yxzG28S0sxDoLkdB2cAkcIDxGWfxxoi62V3IFFkXs5XHcQuTXGLjJymDv+824SaEIid+ePliW4imDUXNJP9gSohTG+zZzNSvKJpipduXd3p+LRkk+HQGMZPHRv2jn24xLpTRkn6uiSVI8xUjA93K04vq3O8CU58/L77GpB5oVZm4GZXr9GgFVI5FsgVi9ps5Glnm4v87d3jTw4sjGy274QbB89Ys43hkUUm4Zu/JlxfZlhkp/o/u+gLuJ/XT65NxTikI9LeY8g5KwM0kA2jQMIL70Iyl+jiJ8f9Venm97toa5/4rKkIqwHkdf2I2f+VRDu2LFtk0VzOjb3mPmAhoOLPWUfLphKJTuIhaI0pdATts9BhJ6bYjj7qLdviRGnx7bOASxCqwPNAGNCXWEVEesMF7Y/Bu56eONFi70pECi3WWgISZjeDhotyKyTusixzQWTWZcBJQ02Ua//JCR0d2Dx33ANEoabwUPaIZ+pLhj8Awaojxf1F0pepWiWCB2DiweXGVfZQyU/EGkBY7tB9HiYNPpNfbAYeqVuIst1DHsYAzPmOhEONrzjKSJBd9bcuX0n7VL15OVRzV5SDmCCS8="
```
Comment (nicolasstucki) [inline .travis.yml]: Is this safe?!?!

Comment (anatoliykmetyuk) [reply .travis.yml]: Yes, that's an encrypted text.

Comment (nicolasstucki) [reply .travis.yml]: Ok, then LGTM

**Comment by anatoliykmetyuk** [comment]
@nicolasstucki can you give me the write permissions to the repo? The API token used in this PR is associated with my account, Travis might not be able to publish if I don't have the write access.

**Pull request merged**